### ISSUE TITLE
Unable to download pretrained imagenet-model

### ISSUE THREAD

**Issue by buxpeng** [ruotianluo/Context-aware-ZSR#2]
hi ，When I download the pretrained imagenet-model according to the README.md, the connection fails all the time. Is the URL in it changed?

**Comment by ruotianluo** [comment]
interseting. Maybe the code breaks.

you can still download by enter in the url: https://docs.google.com/uc?export=download&id=1wHSvusQ1CiEMc5Nx5R8adqoHQjIDWXl1

you can replace id with other ids in that python file.

**Comment by buxpeng** [comment]
Thank you, there is another question, in which folder do I need to put the model downloaded according to the link? Download pretrained model from link in README.md

**Comment by ruotianluo** [comment]
a new directory called pretrained_model under the repo dir.

**Comment by buxpeng** [comment]
![image](https://user-images.githubusercontent.com/53242456/74851558-5ecd9880-5376-11ea-947b-0b56bb371937.png)
Sorry, I mean you reproduce the model in your paper, not pretrained imagenet-model，Is for example this  we_relt_geo_sc.pth  file .

**Comment by ruotianluo** [comment]
a dir called pretrained.
Checkout scripts/reproduce for clue.

**Comment by buxpeng** [comment]
thank you very much！

**Comment by buxpeng** [comment]
Hello, please disturb you again，When I run the test file, it shows that there is no the tagging_eval.pkl file,Does this file need to be downloaded? Where is it if downloaded? Thank you！

**Comment by ruotianluo** [comment]
no. it's the output of tools/test_net.py. i may put the path wrong. do you mind search a little bit where the tagging_eval.pkl is?

**Comment by buxpeng** [comment]
![image](https://user-images.githubusercontent.com/53242456/74897760-0678b400-53d3-11ea-8e55-32b7753c89f9.png)
sorry，i  didn't find where the file was.

**Comment by ruotianluo** [comment]
Can you show me the full error log. And if possible can you run line by line in the script; and also what's your version of pytorch.

**Comment by buxpeng** [comment]
Do you have to use cuda to run? pytorch version is 1.0

**Comment by ruotianluo** [comment]
Yes, cuda is needed.

**Comment by buxpeng** [comment]
i  know ，thank you very much！

**Comment by buxpeng** [comment]
![image](https://user-images.githubusercontent.com/53242456/75030882-0c0ff000-54e0-11ea-9586-1c5e5c9e4b5d.png)
hi,When I run the test file （bash scripts/test/we_infer.sh).I got the above error.These two parameters load_ckpt and load_detectron in the run command are None,Can you look at it for me？

**Comment by ruotianluo** [comment]
You should run scripts/reproduce

**Comment by buxpeng** [comment]
![image](https://user-images.githubusercontent.com/53242456/75093704-181eaf00-55bf-11ea-95d6-4733ba436b88.png)
Hi,Do you need both python2 and python3? I'm installing python 3. Do commands like（In python 2 python lib/datasets/vg/convert_from_bansal.py） need to be executed?

**Comment by ruotianluo** [comment]
I used a conda environment to have both py2 and py3

**Comment by buxpeng** [comment]
I  completed （ Convert from bansal train test split ）using python2，but later compilation fails,My machine is ubantu18  cuda 9.0  gcc have 4.8 5.4 6 7,gcc version i have tried，Can you show it to me?
![image](https://user-images.githubusercontent.com/53242456/75106533-b01ca700-5658-11ea-95ee-e0fe3e910182.png)

**Comment by buxpeng** [comment]
sorry ,gcc have 4.8 5.5 6 7

**Comment by ruotianluo** [comment]
There is no change in this part compared to the original Detectron.pytorch. I suggest you go there and search for the solution.

**Comment by buxpeng** [comment]
Can you tell me which version of cuda and GCC you are using?

**Comment by ruotianluo** [comment]
gcc4.9.4 and cuda10

**Comment by buxpeng** [comment]
hi,I want to ask where this weight is downloaded？
![image](https://user-images.githubusercontent.com/53242456/79069550-b72f6f80-7d01-11ea-9e7a-7cff263026a3.png)

**Comment by ruotianluo** [comment]
Hi, I didn't provide that weight; I only provided the weight after conversion.

**Comment by buxpeng** [comment]
When I run this test file, I can't find the file. Can I ask if this file can be downloaded?
![image](https://user-images.githubusercontent.com/53242456/79069680-afbc9600-7d02-11ea-808f-a2de31c6ffe6.png)

![image](https://user-images.githubusercontent.com/53242456/79069677-a3d0d400-7d02-11ea-8477-795e7247158c.png)

**Comment by ruotianluo** [comment]
Have you trained the model by yourself?

**Comment by buxpeng** [comment]
no ,I am not going to train, I want to use the trained ones directly

**Comment by ruotianluo** [comment]
https://github.com/ruotianluo/Context-aware-ZSR#to-reproduce-numbers-in-the-paper

You need to run the scripts in scripts/reproduce.

**Comment by buxpeng** [comment]
Does everything need to be running? I have run the two inside
![image](https://user-images.githubusercontent.com/53242456/79070113-11323400-7d06-11ea-84db-bc772cde8002.png)

**Comment by ruotianluo** [comment]
It depends on what you want?

**Comment by buxpeng** [comment]
I want to run this
![image](https://user-images.githubusercontent.com/53242456/79070400-2dcf6b80-7d08-11ea-8b58-9aa9090e3a85.png)

**Comment by ruotianluo** [comment]
that's gcn_infer and sync_infer

**Comment by buxpeng** [comment]
I have run gcn_infer in scripts/reproduce，But when I run this bash scripts/test/detection_gcn.sh  ，Still without this weight ckpt/model_final_gcn_wn.pth 
![image](https://user-images.githubusercontent.com/53242456/79071208-fadba680-7d0c-11ea-9c80-b701a26fa709.png)

**Comment by ruotianluo** [comment]
Sorry, my bad, now I understand. 
replace the load_ckpt in detection_gcn with the load_ckpt in gcn_infer
same for sync.

**Comment by buxpeng** [comment]
Oh thanks，This is the result of my running  bash scripts/test/we_infer.sh，Can you tell me what these numbers represent?
![image](https://user-images.githubusercontent.com/53242456/79071669-90783580-7d0f-11ea-8db9-f0d1e917199f.png)

**Comment by ruotianluo** [comment]
It corresponds to Table 1 GCN+Context row.

**Comment by buxpeng** [comment]
thanks,I want to put a picture now, and then use this zero sample to identify what I have seen and what I have n’t seen. Which file do I need to run?

**Comment by ruotianluo** [comment]
Not sure what you mean.

**Comment by buxpeng** [comment]
How can the 7794 pictures in the test set be modified into my own pictures?

**Comment by ruotianluo** [comment]
I haven't done this before. My guess is you need to create a coco annotation type json file for your own images.

If you want to test a single image. You may need to dig around. It shouldn't be too complicated, but it may take a few hours.

**Comment by buxpeng** [comment]
If I only want to test one picture, what do I need to modify?

**Comment by ruotianluo** [comment]
Try if you can extend from this function:
https://github.com/ruotianluo/Context-aware-ZSR/blob/12c1d08ee0e85e38243528127aed79f2efad05b3/lib/core/test_engine.py#L222

But you still need to be able to run edgebox.

**Comment by buxpeng** [comment]
![image](https://user-images.githubusercontent.com/53242456/79226097-69953d00-7e90-11ea-9d10-2e8276d111ab.png)
hi，Can this file be replaced? Is it necessary to install matlab to run this?

**Comment by ruotianluo** [comment]
Unfortunately, it's necessary

**Comment by buxpeng** [comment]
oh,thanks,Is it okay to follow this instruction?
![image](https://user-images.githubusercontent.com/53242456/79291779-207dd100-7f02-11ea-8873-8959c3d159be.png)

**Comment by ruotianluo** [comment]
Yes.

**Comment by buxpeng** [comment]
Now need to install matlab, and then install this Matlab Toolbox?

**Comment by ruotianluo** [comment]
I think so.

**Comment by buxpeng** [comment]
hi,Matlab has been installed and toolbox（3.5.0） has been downloaded, but I did not understand how to do this step. Can you help me?
![image](https://user-images.githubusercontent.com/53242456/79413600-f5b37b80-7fda-11ea-8b3d-2dd86932df69.png)

**Comment by buxpeng** [comment]
![image](https://user-images.githubusercontent.com/53242456/79579175-d6f7d680-80f9-11ea-888f-04316d26a173.png)
![image](https://user-images.githubusercontent.com/53242456/79579275-f42ca500-80f9-11ea-83d9-d44acb9ac1c2.png)
hi,This score is not very clear, why some are greater than 1?

**Comment by ruotianluo** [comment]
I think it’s because the scores after the Crf is not normalized 

Ruotian Luo

**Comment by buxpeng** [comment]
Ok，I tested a picture. The category_id in the generated bbox_vgbansalone_test_results.json is the last category detected with zero ZSR, right?How does this category_id correspond to the real label?

![image](https://user-images.githubusercontent.com/53242456/79583576-dd894c80-80ff-11ea-89f9-33ca95edcadd.png)
![image](https://user-images.githubusercontent.com/53242456/79583638-ed089580-80ff-11ea-85f1-6e23f721279b.png)

**Comment by ruotianluo** [comment]
It should be in some json file in the datasets folder.

**Comment by buxpeng** [comment]
Ok, can you tell me where you can normalize the value after Crf?

**Comment by ruotianluo** [comment]
I am not sure. In fact from https://github.com/ruotianluo/Context-aware-ZSR/blob/master/lib/modeling/rel_heads.py#L135, it seems the scores should be somewhat normalized.

**Comment by ZY123-GOOD** [comment]
> interseting. Maybe the code breaks.
> 
> you can still download by enter in the url: https://docs.google.com/uc?export=download&id=1wHSvusQ1CiEMc5Nx5R8adqoHQjIDWXl1
> 
> you can replace id with other ids in that python file.

Hello, could you please refresh this download link? Best wishes.

**Comment by ruotianluo** [comment]
> > interseting. Maybe the code breaks.
> > you can still download by enter in the url: https://docs.google.com/uc?export=download&id=1wHSvusQ1CiEMc5Nx5R8adqoHQjIDWXl1
> > you can replace id with other ids in that python file.
> 
> Hello, could you please refresh this download link? Best wishes.

https://drive.google.com/drive/folders/0B7fNdx_jAqhtdXR1NUN4NkZwS00?resourcekey=0-4-sXYhJKBsRD8sohsPfDvA&usp=sharing this folder should have all the weights.
